src/lib/optimizer/strategy/join_predicate_ordering_rule.cpp

```diff
@@ -0,0 +1,70 @@
+#include "join_predicate_ordering_rule.hpp"
+
+#include <algorithm>
+
+#include "cost_estimation/abstract_cost_estimator.hpp"
+#include "expression/abstract_predicate_expression.hpp"
+#include "expression/expression_utils.hpp"
+#include "expression/logical_expression.hpp"
+#include "logical_query_plan/lqp_utils.hpp"
+#include "statistics/cardinality_estimator.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+void JoinPredicateOrderingRule::apply_to(const std::shared_ptr<AbstractLQPNode>& root) const {
+  visit_lqp(root, [&](const auto& node) {
+    // Check if this is a multi predicate join.
+    if (node->type != LQPNodeType::Join || node->node_expressions.size() <= 1) {
+      return LQPVisitation::VisitInputs;
+    }
+
+    const auto join_mode = std::static_pointer_cast<JoinNode>(node)->join_mode;
+
+    DebugAssert(cost_estimator, "JoinOrderingRule requires cost estimator to be set");
+    const auto caching_cardinality_estimator = cost_estimator->cardinality_estimator->new_instance();
+
+    // Estimate join selectivity of a predicate by creating a new join node joining only on that one predicate and
+    //  estimating that join node's cardinality.
+    //  “The join selectivity is the ratio “number of tuples in the result/number of tuples in the cartesian product.”
```

Comment:
Unmatched quotes. That's why gave me nightmares last night :D 
Also, use regular quotes, please